modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/BusyExecutor.java

```diff
@@ -135,6 +194,25 @@ public boolean busyRun(Runnable r) {
      * @param r Task to execute.
      */
     public void execute(Runnable r) {
-        pool.execute(() -> busyRun(r));
+        GridBusyLock lock = busyLock;
+
+        pool.execute(() -> busyRun(r, lock));
```

Comment:
```suggestion
        submit(r);
```

Reply:
I'd say 
-2 methods (26 lines of code) that bring no value
+1 light object. it doen't look like it is not on hot path.